src/SpectatorView.Unity/Assets/SpectatorView.Editor/Scripts/StateSynchronizationPerformanceWindow.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using UnityEngine;
+using UnityEditor;
+using System.ComponentModel;
+
+namespace Microsoft.MixedReality.SpectatorView.Editor
+{
+    [Description("Performance")]
+    internal class StateSynchronizationPerformanceWindow : CompositorWindowBase<StateSynchronizationPerformanceWindow>
+    {
+        private static readonly string appIPAddressKey = $"{nameof(StateSynchronizationPerformanceWindow)}.{nameof(appIPAddress)}";
+        private string appIPAddress;
+        private const int globalSettingsButtonWidth = 220;
+        private Vector2 scrollPosition;
+
+        [MenuItem("Spectator View/Performance", false, 3)]
+        public static void ShowCalibrationRecordingWindow()
+        {
+            ShowWindow();
+        }
+
+        protected override void OnEnable()
+        {
+            base.OnEnable();
+            appIPAddress = PlayerPrefs.GetString(appIPAddressKey, "localhost");
+        }
+
+        protected override void OnDisable()
+        {
+            base.OnDisable();
+            PlayerPrefs.SetString(appIPAddressKey, appIPAddress);
+        }
+
+        private void OnGUI()
+        {
+            EditorGUILayout.BeginVertical();
+            {
+                if (GUILayout.Button(new GUIContent("Open performance settings prefab", "Opens the prefab that defines global performance settings."), GUILayout.Width(globalSettingsButtonWidth)))
+                {
+                    StateSynchronizationMenuItems.EditGlobalPerformanceParameters();
+                }
+
+                if (!EditorApplication.isPlaying)
+                {
+                    var observer = FindObjectOfType<StateSynchronizationObserver>();
```

Comment:
>FindObjectOfType<StateSynchronizationObserver>(); [](start = 35, length = 49)

I think this could be StateSynchronizationObserver.Instance

Reply:
Good call, our implementation of singleton doesn't instantiate things when they don't exist. I've seen varying implementations.